### PR TITLE
fix: better print propertylookup nodes

### LINKED ISSUE #373: Better output `propertylookup` and `offsetlookup`
Input:
```php
$SomeVeryLongUpperCaseConstant->someVeryLongCallExpression()->some_very_long_member_expression;
$weNeedToReachTheEightyCharacterLimitXXXXXXXXXXXXXXXXX->someNode
->childrenInAnArray[0];
$superSupersuperSupersuperSupersuperSupersuperSuperLong->exampleOfOrderOfGetterAndSetterReordered;
$superSupersuperSupersuperSupersuperSupersuperSuperLong->exampleOfOrderOfGetterAndSetterReordered[0];
```
Output:
```php
$SomeVeryLongUpperCaseConstant->someVeryLongCallExpression()->
    some_very_long_member_expression;
$weNeedToReachTheEightyCharacterLimitXXXXXXXXXXXXXXXXX->someNode->
    childrenInAnArray[0];
$superSupersuperSupersuperSupersuperSupersuperSuperLong->
    exampleOfOrderOfGetterAndSetterReordered;
$superSupersuperSupersuperSupersuperSupersuperSuperLong->
    exampleOfOrderOfGetterAndSetterReordered[0];
```

Also need add tests for `offsetlookup`

### PR DESCRIPTION
**Pull request by evilebottnawi**
fixes #373

Port https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L560 (maybe need more check in `shouldInline`, if somebody have idea put example here, if not we can add more nodes after new issues)

### CODE REVIEW

File: tests/class/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -616,7 +616,8 @@ class test
         $customer = (object) ['name' => 'Bob'];
         $job = (object) ['customer' => $customer];
 
-        return "The customer for that job, {$job->customer->name} has an error that shows up after the line gets waaaaay toooo long.";
+        return "The customer for that job, {$job->customer
+            ->name} has an error that shows up after the line gets waaaaay toooo long.";
```
Comment (evilebottnawi) [inline tests/class/__snapshots__/jsfmt.spec.js.snap]: Should be fixed in other PR, like here https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAnOMCuqoAIAGAKgBZy5iYDOMEAtnKrgGYSMzECGMuAVhAEYAaXABJgIvvwB0FanQZSoHegF9cnSrg54GqVrnZdclYhADumzAActTeG1K4ANgEsoZAOYZNZjn78AngYQIRDO0B5S+ADcIIIgEFYwLtCUyKAcqHpmAAqZCGkoHABuEC4AJnEg-KgcYADWGADKVnVuHsgwqJhw8W6UDDA5tR60HMhMHE4D8TyUAB4AQrUNzcpwADJucBNTMyCtqAOoyNUc-AFO0FVWqG4wAOoV7MgAHAAM8bcQAw+1VqdbnBjsUdvF0ABHTAudDDDijcZISbTXogAa0FydbqoyjtJxwACKmAg8F2KPiMHOT3KLyQACYKbUXK4oB4AMJ0ManKDQMEgKhwQjnQrIgYqFRAA

File: tests/switch/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -316,7 +316,8 @@ switch (
 }
 
 switch (1) {
-    case $exception->veryVeryVeryVeryVeryVeryVeryVeryVeryLongMethod()->veryVeryVeryVeryVeryVeryVeryLongProperty:
+    case $exception->veryVeryVeryVeryVeryVeryVeryVeryVeryLongMethod()
+        ->veryVeryVeryVeryVeryVeryVeryLongProperty:
```
Comment (evilebottnawi) [inline tests/switch/__snapshots__/jsfmt.spec.js.snap]: https://github.com/prettier/prettier/issues/4735#issuecomment-399494589

File: tests/member_chain/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -49,12 +49,35 @@ $weNeedToReachTheEightyCharacterLimitXXXXXXXXXXXXXXXXX->someNode
...
+$weNeedToReachTheEightyCharacterLimitXXXXXXXXXXXXXXXXX->someNode
+    ->childrenInAnArray[0];
 $superSupersuperSupersuperSupersuperSupersuperSuperLong->exampleOfOrderOfGetterAndSetterReordered;
-$superSupersuperSupersuperSupersuperSupersuperSuperLong->exampleOfOrderOfGetterAndSetterReordered[0];
+$superSupersuperSupersuperSupersuperSupersuperSuperLong
+    ->exampleOfOrderOfGetterAndSetterReordered[0];
+$superSupersuperSupersuperSupersuperSupersuperSuperLong->{$exampleOfOrderOfGetterAndSetterReordered};
+$superSupersuperSupersuperSupersuperSupersuperSuperLong::$exampleOfOrderOfGetterAndSetterReordered;
+$superSupersuperSupersuperSupersuperSupersuperSuperLong::$exampleOfOrderOfGetterAndSetterReordered[0];
+$superSupersuperSupersuperSupersuperSupersuperSuperLong = (new SuperSuperSuperSuperSuperSuperLongClassName())
```
Comment (mgrip) [inline tests/member_chain/__snapshots__/jsfmt.spec.js.snap]: shouldn't this break at the `=` first?

**Comment by mgrip** [comment]
Also I'm not sure if one is better than the other, but just as a heads up the issue you originally opened was requesting output like this

```php
$SomeVeryLongUpperCaseConstant->someVeryLongCallExpression()->
    some_very_long_member_expression;
```

This appears to do

```php
$SomeVeryLongUpperCaseConstant->someVeryLongCallExpression()
    ->some_very_long_member_expression;
```

File: tests/member_chain/__snapshots__/jsfmt.spec.js.snap (hunk 3 above)
Comment (evilebottnawi) [reply tests/member_chain/__snapshots__/jsfmt.spec.js.snap]: @mgrip js prettier don't break, i do same logic, we can change this if a lot of people vote for break

**Comment by evilebottnawi** [comment]
@mgrip let's open new issue around this, out of scope this PR

**Comment by mgrip** [comment]
fine by me! good to merge 👍

**Pull request merged**